OpenSim/Simulation/SimbodyEngine/EllipsoidJoint.cpp

```diff
@@ -99,40 +99,22 @@ void EllipsoidJoint::setEllipsoidRadii(const Vec3& radii)
     set_radii_x_y_z(radii);
 }
 
-//=============================================================================
+//==============================================================================
 // SCALING
-//=============================================================================
-//_____________________________________________________________________________
-/**
- * Scale a joint based on XYZ scale factors for the bodies.
- *
- * @param aScaleSet Set of XYZ scale factors for the bodies.
- * @todo Need to scale transforms appropriately, given an arbitrary axis.
- */
-void EllipsoidJoint::scale(const ScaleSet& aScaleSet)
+//==============================================================================
+void EllipsoidJoint::scale(const SimTK::State& s, const ScaleSet& scaleSet)
 {
-    Vec3 scaleFactors(1.0);
-
-    // Joint knows how to scale locations of the joint in parent and on the body
-    Super::scale(aScaleSet);
-
-    // SCALING TO DO WITH THE PARENT BODY -----
-    // Joint kinematics are scaled by the scale factors for the
-    // parent body, so get those body's factors
-    const string& parentName = getParentFrame().getName();
-    for (int i=0; i<aScaleSet.getSize(); i++) {
-        Scale& scale = aScaleSet.get(i);
-        if (scale.getSegmentName()==parentName) {
-            scale.getScaleFactors(scaleFactors);
-            break;
-        }
-    }
+    Super::scale(s, scaleSet);
 
-    SimTK::Vec3& ellipsoidRadii = upd_radii_x_y_z();
-    for(int i=0; i<3; i++){ 
-        // Scale the size of the mobilizer
-        ellipsoidRadii[i] *= scaleFactors[i];
-    }
+    // Get scale factors for base frame (if an entry for the base frame exists).
+    const int idx = scaleSet.getIndexBySegmentName(getParentFrame()
+                                                   .findBaseFrame().getName());
+    if (idx < 0)
+        return;
+    const Vec3& scaleFactors = scaleSet[idx].getScaleFactors();
+
+    //TODO: Need to scale transforms appropriately, given an arbitrary axis.
+    upd_radii_x_y_z() = get_radii_x_y_z().elementwiseMultiply(scaleFactors);
```

Comment:
If you want to get really fancy you could do `upd_radii_x_y_z().elementwiseMultiplyInPlace(scaleFactors);`

Reply:
<pre>Error: class "SimTK::Vec<3, SimTK::Real, 1>" has no member "elementwiseMultiplyInPlace"</pre>
😕